### PR TITLE
reduce number of lines

### PR DESCRIPTION
**Pull request by croumegous**
Naively reduce number of lines (43 less lines), by using walrus operator, multi assignment, and removing new line for simple `if` and `for` statement

### CODE REVIEW

File: tinygrad/jit.py [hunk 1]
```diff
@@ -8,10 +8,8 @@
 
 class TinyJit:
   def __init__(self, fxn:Callable):
-    self.fxn = fxn
```
Comment (geohot) [inline tinygrad/jit.py:11]: This should have types

File: tinygrad/lazy.py [hunk 2]
```diff
@@ -105,17 +104,14 @@ def __init__(self, device:str, shape:Union[ShapeTracker, Tuple[int, ...]], optyp
...
+    for x in get_buffers(op): x.children.add(self)
+    if not LAZY: self.realize()
 
   def __repr__(self): return f"<LB {self.shape} op:{self.op.op if self.realized is None else 'realized'}>"
 
   # this produces a device buffer
   def realize(self:LazyBuffer, required_device=None) -> DeviceBuffer:
-    if required_device is not None:
-      assert required_device == self.device
+    if required_device is not None: assert required_device == self.device
```
Comment (geohot) [inline tinygrad/lazy.py:114]: This should be rewritten as only the assert

File: tinygrad/ops.py [hunk 3]
```diff
@@ -124,8 +124,7 @@ def optimize_local_size(self, bufs) -> List[int]:
   def lower(self, bufs) -> List[RawBuffer]: return [x.raw() for i,x in enumerate(bufs) if x is not None and i not in self.bufs_to_delete]
   def __call__(self, bufs):
     if getenv("OPTLOCAL") and self.global_size is not None and self.local_size is None: self.local_size = self.optimize_local_size(bufs)
-    et = self.clprg(self.global_size, self.local_size, *bufs, wait=DEBUG>=2)
-    if et is not None: GlobalCounters.time_sum_s += et
+    if et := self.clprg(self.global_size, self.local_size, *bufs, wait=DEBUG>=2): GlobalCounters.time_sum_s += et
```
Comment (geohot) [inline tinygrad/ops.py:127]: Neat!

File: tinygrad/shape/__init__.py [hunk 4]
```diff
@@ -29,8 +29,7 @@ def expr_node(self, idx=None, offset=0):
     ret = [Variable.num(self.offset+offset)]
     acc = 1
     for d,s in self.shape_strides[::-1]:
-      if d != 1 and s != 0:
-        ret.append(((idx//acc)%d)*s)
+      if d != 1 and s != 0: ret.append(((idx//acc)%d)*s)
```
Comment (geohot) [inline tinygrad/shape/__init__.py:32]: You don't need this if statement actually, symbolic should deal fine

File: tinygrad/nn/optim.py [hunk 5]
```diff
@@ -69,7 +67,7 @@ def __init__(self, params : List[Tensor], lr=0.001, b1=0.9, b2=0.999, eps=1e-8):
     self.v = [Tensor.zeros(*t.shape, device=params[0].device, requires_grad=False) for t in self.params]
 
   def step(self) -> None:
-    self.t = self.t + 1
+    self.t += 1
```
Comment (geohot) [inline tinygrad/nn/optim.py:70]: So this actually changes behavior. But I think it was wrong before...

File: tinygrad/ops.py (hunk 3 above)
Comment (croumegous) [reply tinygrad/ops.py:127]: The scope is not limited to the if block, so there are no issues with it.
```python
>>> a = 2
>>> if a := 1 :
...     print(a)
...
1
>>> a
1


>>> if b := 2:
...     print(b)
...
2
>>> b
2

```

**Comment by croumegous** [comment]
@geohot all comments should have been addressed and resolved

File: tinygrad/jit.py [hunk 6]
```diff
@@ -8,10 +8,10 @@
...
-    self.cnt = 0
+    self.fxn: Callable = fxn
+    self.cnt: int = 0
     self.jit_cache : List[Tuple[Callable, Any]] = []  # TODO: Any should be List[RawBuffer], but this fails
-    self.ret = None
+    self.ret: Optional[Tensor] = None
```
Comment (geohot) [inline tinygrad/jit.py:14]: This doesn't have to be a Tensor (fxn can return anything), Any is fine

Comment (geohot) [reply tinygrad/jit.py:14]: Also, match the "var : type" convention, not "var: type"

I'm open to changing this across the codebase if the other is more common

**Comment by geohot** [comment]
If tests pass + address type comment, it's good to merge.

File: tinygrad/nn/optim.py (hunk 5 above)
Comment (geohot) [reply tinygrad/nn/optim.py:70]: Yea, this is breaking the tests. You can either revert it, or investigate!

**Comment by croumegous** [comment]
Type updated to Any.
`var: int = 0`  is indeed more common/idiomatic than `var : int = 0`, I'll make a PR to apply this change everywhere if it's ok.

I reverted `self.t += 1` to fix optim test, I might investigate later.

**Comment by geohot** [comment]
I'm fine with changing to `var: int` if you make sure the linter enforces it.

**Comment by geohot** [comment]
Merged, thanks!

**Pull request merged**